packages/logger/src/utils/sonic-boom.ts

```diff
@@ -1,28 +1,17 @@
 import { OgmaStream } from '@ogma/common';
+import SonicBoom from 'sonic-boom';
 
-const noop = () => {}; // eslint-disable-line @typescript-eslint/no-empty-function
+const noop = (_args?: any) => true; // eslint-disable-line @typescript-eslint/no-empty-function
 
 /**
  * thanks pinojs
  * ref: https://github.com/pinojs/pino/blob/27d2ab8b58e64547fd24864c2f21ac898f4752c4/lib/tools.js#L219
  */
-function buildSafeSonicBoom(opts) {
-  /* eslint-disable @typescript-eslint/no-var-requires */
-  const onExit = require('on-exit-leak-free');
-  const SonicBoom = require('sonic-boom');
-  const { isMainThread } = require('child_process');
-
-  const stream = new SonicBoom(opts);
+function buildSafeSonicBoom(opts: { fd: typeof process['stdout']['fd'] }) {
+  // the ?? 1 is here to handle worker threads like using vitest
+  const stream = new SonicBoom({ fd: opts.fd ?? 1 });
 
   stream.on('error', filterBrokenPipe);
-  // if we are sync: false, we must flush on exit
-  if (!opts.sync && isMainThread) {
```

Comment:
Making a note for myself more than anything:

We don't pass the `opts.sync` option in first place, and `child_process` does not have an `isMainThread` property, so this will always evaluate to `!false && false` i.e. it will always be `false`. As such, we can remove a good bit of this file and clean things up a bit